llvm/lib/CodeGen/AsmPrinter/DebugHandlerBase.cpp

```diff
@@ -215,22 +215,26 @@ bool DebugHandlerBase::isUnsignedDIType(const DIType *Ty) {
     return isUnsignedDIType(DTy->getBaseType());
   }
 
-  auto *BTy = cast<DIBasicType>(Ty);
-  unsigned Encoding = BTy->getEncoding();
-  assert((Encoding == dwarf::DW_ATE_unsigned ||
-          Encoding == dwarf::DW_ATE_unsigned_char ||
-          Encoding == dwarf::DW_ATE_signed ||
-          Encoding == dwarf::DW_ATE_signed_char ||
-          Encoding == dwarf::DW_ATE_float || Encoding == dwarf::DW_ATE_UTF ||
-          Encoding == dwarf::DW_ATE_boolean ||
-          Encoding == dwarf::DW_ATE_complex_float ||
-          (Ty->getTag() == dwarf::DW_TAG_unspecified_type &&
-           Ty->getName() == "decltype(nullptr)")) &&
-         "Unsupported encoding");
-  return Encoding == dwarf::DW_ATE_unsigned ||
-         Encoding == dwarf::DW_ATE_unsigned_char ||
-         Encoding == dwarf::DW_ATE_UTF || Encoding == dwarf::DW_ATE_boolean ||
-         Ty->getTag() == dwarf::DW_TAG_unspecified_type;
+  if (auto *BTy = dyn_cast<DIBasicType>(Ty)) {
+    unsigned Encoding = BTy->getEncoding();
+    assert((Encoding == dwarf::DW_ATE_unsigned ||
+            Encoding == dwarf::DW_ATE_unsigned_char ||
+            Encoding == dwarf::DW_ATE_signed ||
+            Encoding == dwarf::DW_ATE_signed_char ||
+            Encoding == dwarf::DW_ATE_float || Encoding == dwarf::DW_ATE_UTF ||
+            Encoding == dwarf::DW_ATE_boolean ||
+            Encoding == dwarf::DW_ATE_complex_float ||
+            (Ty->getTag() == dwarf::DW_TAG_unspecified_type &&
+             Ty->getName() == "decltype(nullptr)")) &&
+           "Unsupported encoding");
+    return Encoding == dwarf::DW_ATE_unsigned ||
+           Encoding == dwarf::DW_ATE_unsigned_char ||
+           Encoding == dwarf::DW_ATE_UTF || Encoding == dwarf::DW_ATE_boolean ||
+           Ty->getTag() == dwarf::DW_TAG_unspecified_type;
+  }
+  // FIXME: the signedness should come from the expression where the type is
+  // used in, not the type itself.
+  return true;
```

Comment:
On which code path dis this originally crash? Just checking whether this still addresses the original issue you wanted to solve.

Reply:
This was originally a DICompositeType with an `DW_TAG_enumeration_type` tag, whose base type was another DICompositeType.